### PR TITLE
Merge OngXinBin-CodeQuality

### PR DESCRIPTION
**Pull request by n3wsoldier**
1) remove magic literal
2) change user guide expected outcome for edit bookmark episode

### CODE REVIEW

**Review by EyoWeiChin — APPROVED** [review]
Nice job refactoring everything, thanks! LGTM!

File: src/main/java/anichan/anime/AnimeStorage.java [hunk 1]
```diff
@@ -22,27 +22,70 @@
...
+    private static final String YEAR_JSON_FIELD = "year";
+    private static final String MONTH_JSON_FIELD = "month";
+    private static final String DAY_JSON_FIELD = "day";
+    private static final String AVERAGE_SCORE_JSON_FIELD = "averageScore";
+    private static final String GENRES_JSON_FIELD = "genres";
+    private static final String DURATION_JSON_FIELD = "duration";
+    private static final String TITLE_JSON_FIELD = "title";
+
+    //Error messages
+    private static final String GENRE_NULL_ERROR = "Genre should not be null.";
```
Comment (ChanJianHao) [inline src/main/java/anichan/anime/AnimeStorage.java:62]: Great work refactoring!

File: src/main/java/anichan/anime/AnimeStorage.java [hunk 2]
```diff
@@ -22,27 +22,70 @@
...
     public ArrayList<Anime> readAnimeDatabase() throws AniException {
-        LOGGER.log(Level.INFO, "Retrieving information from DataSource.");
+        LOGGER.log(Level.INFO, ANIME_STORAGE_RETRIEVE_MESSAGE);
         ArrayList<Anime> animeDataList = new ArrayList<>();
-        for (int i = 1; i < 6; i++) {
-            LOGGER.log(Level.INFO, "Currently extracting from /AniListData/AniList-Data" + i + ".json");
-            String fileData = getDataFromJarFile("/AniListData/AniList-Data" + i + ".json");
-            LOGGER.log(Level.INFO, "Extraction of /AniListData/AniList-Data" + i + ".json successful");
-            LOGGER.log(Level.INFO, "Parsing Json data.");
+        for (int i = FIRST_FILE_INDEX; i < LAST_FILE_INDEX; i++) {
```
Comment (ChanJianHao) [inline src/main/java/anichan/anime/AnimeStorage.java:80]: Hmmmmmmmmmmm looks good, maybe.

File: src/main/java/anichan/bookmark/Bookmark.java [hunk 3]
```diff
@@ -9,6 +9,10 @@
  * Represent the bookmark of a workspace.
  */
 public class Bookmark {
+    private static final String NOTES_EMPTY_MESSAGE = "\tNotes is empty.. :(";
+    private static final String LIST_EMPTY_MESSAGE = "\tUhh.. It's empty.. :(";
+    private static final int EMPTY_SIZE = 0;
+    private static final int ONE_BASED_OFFSET = 1;
```
Comment (ChanJianHao) [inline src/main/java/anichan/bookmark/Bookmark.java:15]: Good use of offset

**Review by ChanJianHao — APPROVED** [review]
LGTM!

**Pull request merged**